### PR TITLE
Add method to extract _elem_crack_origin_direction_map.

### LINKED ISSUE #12435: Add method to extract _elem_crack_origin_direction_map from XFEM.C for use in other MOOSE objects
## Rationale

In a postprocessor I am developing in Bison I require access to the information contained in the _elem_crack_origin_direction_map populated in XFEM.C. 

## Description
Add a method to return the data in the map such that it can be used by other MOOSE objects.

## Impact

None. Grants access to already existing data in the map.

### PR DESCRIPTION
**Pull request by gambka**
Adding method to extract data in an existing map in XFEM.C for use in other MOOSE objects.

Closes #12435.

### CODE REVIEW

**Comment by gambka** [comment]
@jiangwen84 @bwspenc

File: modules/xfem/include/base/XFEM.h [hunk 1]
```diff
@@ -244,6 +244,11 @@ class XFEM : public XFEMInterface
                         EFAElement3D * CEMElem,
                         std::vector<std::vector<Point>> & frag_faces) const;
 
+  virtual std::map<const Elem *, std::vector<Point>> getCrackTipOriginMap() const
```
Comment (jiangwen84) [inline modules/xfem/include/base/XFEM.h]: I think this function does not need to be virtual.

**Comment by permcody** [comment]
The problem with this PR is that there's a new method (just an accessor) added, but no test and the method is not used in MOOSE nor the modules...   

I could remove this tomorrow and still pass all MOOSE and modules testing. I would really prefer that there was a test added (even if that means making a pure test object in the XFEM module, even if that object is a little contrived) that requires this API.

**Comment by gambka** [comment]
Where would be the best place to store the .md documentation for these pure test objects? There is no `doc` directory beside `include` and `src` in `test/`.  I thought the purpose of pure test objects is to for testing and not to be used by users.  I haven't seen pure test objects documented elsewhere as of yet.  Is there a special grouping on the documentation webpage for these objects to ensure users are aware to not use them?

**Comment by permcody** [comment]
> Where would be the best place to store the .md documentation for these pure test objects? There is no doc directory beside include and src in test/. 

Don't document the pure test objects. That's our current thinking.

> I thought the purpose of pure test objects is to for testing and not to be used by users. I haven't seen pure test objects documented elsewhere as of yet. 

Right!

Is there a special grouping on the documentation webpage for these objects to ensure users are aware to not use them?

Nope! Users can use pure test objects if they really want. There's the "--allow-test-objects" flag that makes them available to instantiate from a normal run. Normally, you won't use this unless you really want to use a "test" object in a production run (whatever that means).

**Comment by gambka** [comment]
The reason I asked is because the testing system failed the documentation because I didn't have .md files for the pure test object.

**Comment by permcody** [comment]
It shouldn't fail. If it does again, I'll have @aeslaughter take a look. Thanks for working on this!

**Review by permcody — CHANGES_REQUESTED** [review]

**Review by permcody — APPROVED** [review]

**Comment by permcody** [comment]
Excellent work Kyle! Thanks for jumping through all these extra hoops.

**Pull request merged**